api/server/router/network/network_routes.go

```diff
@@ -56,6 +56,10 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 		return err
 	}
 
+	if vars["id"] == "" {
+		return libnetwork.ErrNoSuchNetwork(vars["id"])
```

Comment:
See the repro case at the top; probably not possible through the CLI, but it's possible to get there when directly using the API